add-chain/contract_addresses.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type AddressData struct {
+	Address string `json:"address"`
+}
+
+var (
+	// Addresses to retrieve from JSON
+	AddressManager                    = "AddressManager"
+	L1CrossDomainMessengerProxy       = "L1CrossDomainMessengerProxy"
+	L1ERC721BridgeProxy               = "L1ERC721BridgeProxy"
+	L1StandardBridgeProxy             = "L1StandardBridgeProxy"
+	L2OutputOracleProxy               = "L2OutputOracleProxy"
+	OptimismMintableERC20FactoryProxy = "OptimismMintableERC20FactoryProxy"
+	SystemConfigProxy                 = "SystemConfigProxy"
+	OptimismPortalProxy               = "OptimismPortalProxy"
+	ProxyAdmin                        = "ProxyAdmin"
+
+	// Addresses to retrieve from chain
+	SuperchainConfig  = "SuperchainConfig"
+	Guardian          = "Guardian"
+	Challenger        = "Challenger"
+	ProxyAdminOwner   = "ProxyAdminOwner"
+	SystemConfigOwner = "SystemConfigOwner"
+)
+
+func readAddressesFromChain(contractAddresses map[string]string, l1RpcUrl string) error {
+	// SuperchainConfig
+	address, err := executeCommand("cast", []string{"call", contractAddresses[OptimismPortalProxy], "superchainConfig()(address)", "-r", l1RpcUrl})
+	address = strings.Join(strings.Fields(address), "") // remove whitespace
+	if err != nil || address == "" || address == "0x" {
+		contractAddresses[SuperchainConfig] = ""
+	} else {
+		contractAddresses[SuperchainConfig] = address
+	}
+
+	// Guardian
+	address, err = executeCommand("cast", []string{"call", contractAddresses[SuperchainConfig], "guardian()(address)", "-r", l1RpcUrl})
```

Comment:
Is it worth factoring this into a `castCall(address string, calldata string)` helper?

Reply:
Refactored here: https://github.com/ethereum-optimism/superchain-registry/pull/204/commits/2fd6532f5eecadaa8fca62395d7301b23a1bc2b4